compiler/modules/front.syntax/parseTypeDesc.bal

```diff
@@ -277,27 +277,35 @@ function parseTypeParam(Tokenizer tok) returns TypeDesc|err:Syntax {
 }
 
 // current token should be "("
-function parseFunctionTypeDesc(Tokenizer tok, string[]? paramNames = ()) returns FunctionTypeDesc|err:Syntax {
+function parseFunctionTypeDesc(Tokenizer tok, FunctionParam[]? fnParams = ()) returns FunctionTypeDesc|err:Syntax {
     // skip "function"
     Position startPos = tok.currentStartPos();
     check tok.expect("(");
-    TypeDesc[] args = [];
+    FunctionTypeParam[] params = [];
     while true {
         if tok.current() == ")" {
             break;
         }
-        args.push(check parseTypeDesc(tok));
+        Position paramStartPos = tok.currentStartPos();
+        TypeDesc td = check parseTypeDesc(tok);
+        string? name = ();
+        Position? namePos = ();
         match tok.current() {
             [IDENTIFIER, var paramName] => {
-                if paramNames != () {
-                    paramNames.push(paramName);
+                name = paramName;
+                namePos = tok.currentStartPos();
+                FunctionParam param = { startPos: paramStartPos, endPos: tok.currentEndPos(), name: <string>name, namePos: <Position>namePos, td };
```

Comment:
Most times if your code has a cast, you are doing it wrong...

You should declare name, namePos in each of the two blocks with the correct type each case.

Reply:
fixed